stdlib/importlib/resources/simple.pyi

```diff
@@ -28,11 +28,31 @@ if sys.version_info >= (3, 11):
         def is_file(self) -> Literal[True]: ...
         def is_dir(self) -> Literal[False]: ...
         @overload
-        def open(self, mode: OpenTextMode = "r", *args: Incomplete, **kwargs: Incomplete) -> TextIOWrapper: ...
+        def open(
+            self,
+            mode: OpenTextMode = "r",
+            *,
+            encoding: str | None = ...,
+            errors: str | None = ...,
+            newline: str | None = ...,
+            line_buffering: bool = ...,
+            write_through: bool = ...,
+        ) -> TextIOWrapper: ...
         @overload
-        def open(self, mode: OpenBinaryMode, *args: Unused, **kwargs: Unused) -> BinaryIO: ...
+        def open(
+            self,
+            mode: OpenTextMode,
+            buffer: _WrappedBuffer,
+            encoding: str | None = ...,
+            errors: str | None = ...,
+            newline: str | None = ...,
+            line_buffering: bool = ...,
+            write_through: bool = ...,
+        ) -> TextIOWrapper: ...
```

Comment:
For text modes, the arguments are just passed on to `TextIOWrapper.__init__()`. That said, I'm not sure if passing a buffer here is really a good idea. But then, the implementation of this method is strange anyway, creating a stream when using the text mode, but then not passing that stream to `TextIOWrapper` ...

Reply:
It was indeed a CPython bug: python/cpython#111775.